equinix/data_source_network_device.go

```diff
@@ -0,0 +1,906 @@
+	"Name":              "Interface name",
+	"Status":            "Interface status (AVAILABLE, RESERVED, ASSIGNED)",
+	"OperationalStatus": "Interface operational status (up or down)",
+	"MACAddress":        "Interface MAC addres",
+	"IPAddress":         "interface IP address",
+	"AssignedType":      "Interface management type (Equinix Managed or empty)",
+	"Type":              "Interface type",
+}
+
+var ecxDeviceUserKeySchemaNames = map[string]string{
+	"Username": "username",
+	"KeyName":  "key_name",
+}
+
+var ecxDeviceUserKeyDescriptions = map[string]string{
+	"Username": "Username associated with given key",
+	"KeyName":  "Reference by name to previously provisioned public SSH key",
+}
+
+var ecxDeviceClusterSchemaNames = map[string]string{
+	"ClusterId":   "cluster_id",
+	"ClusterName": "cluster_name",
+	"NumOfNodes":  "num_of_nodes",
+	"Node0":       "node0",
+	"Node1":       "node1",
+}
+
+var ecxDeviceClusterDescriptions = map[string]string{
+	"ClusterId":   "The id of the cluster",
+	"ClusterName": "The name of the cluster device",
+	"NumOfNodes":  "The number of nodes in the cluster",
+	"Node0":       "An object that has node0 details",
+	"Node1":       "An object that has node1 details",
+}
+
+var ecxDeviceClusterNodeSchemaNames = map[string]string{
+	"VendorConfiguration": "vendor_configuration",
+	"LicenseFileId":       "license_file_id",
+	"LicenseToken":        "license_token",
+	"UUID":                "uuid",
+	"Name":                "name",
+}
+
+var ecxDeviceClusterNodeDescriptions = map[string]string{
+	"VendorConfiguration": "An object that has fields relevant to the vendor of the cluster device",
+	"LicenseFileId":       "License file id. This is necessary for Fortinet and Juniper clusters",
+	"LicenseToken":        "License token. This is necessary for Palo Alto clusters",
+	"UUID":                "The unique id of the node",
+	"Name":                "The name of the node",
+}
+
+var ecxDeviceVendorConfigSchemaNames = map[string]string{
+	"Hostname":       "hostname",
+	"AdminPassword":  "admin_password",
+	"Controller1":    "controller1",
+	"ActivationKey":  "activation_key",
+	"ControllerFqdn": "controller_fqdn",
+	"RootPassword":   "root_password",
+}
+
+var ecxDeviceVendorConfigDescriptions = map[string]string{
+	"Hostname":       "Hostname. This is necessary for Palo Alto, Juniper, and Fortinet clusters",
+	"AdminPassword":  "The administrative password of the device. You can use it to log in to the console. This field is not available for all device types",
+	"Controller1":    "System IP Address. Mandatory for the Fortinet SDWAN cluster device",
+	"ActivationKey":  "Activation key. This is required for Velocloud clusters",
+	"ControllerFqdn": "Controller fqdn. This is required for Velocloud clusters",
+	"RootPassword":   "The CLI password of the device. This field is relevant only for the Velocloud SDWAN cluster",
+}
+
+func createDataSourceNetworkDeviceSchema() map[string]*schema.Schema {
```

Comment:
Schema definitions were also copied and renamed from the Network Device Resource, but also I had to modify all or nearly all of the elements to be `Computed: true`, remove validators, etc.  Again not sure if this exactly the right way to do this, but Terraform definitely didn't like elements with 'Required: true' or validators and similar fields specified.